app/api/Torrent.js

```diff
@@ -48,8 +48,21 @@ export default class Torrent {
 
     this.engine.add(magnetURI, { path: cacheLocation }, torrent => {
       const server = torrent.createServer();
-      server.listen(port);
-      this.server = server;
+      const thisRef = this;
+
+      detect(this.port).then((_port) => {
+        if (thisRef.port !== _port) {
+          console.error(`Port ${thisRef.port} is already in use. Using ${_port} instead.`);
+        }
+
+        thisRef.port = _port;
+        server.listen(thisRef.port);
+        thisRef.server = server;
+        return;
+      }).catch((err) => {
```

Comment:
There should be an eslint rule for this but multiple chained calls should always be broken to a new line:
ex:
```js
some
  .then(res => res.json())
  .catch(err => console.log(res))
```